XTS/WS-C/dev/src/com/arjuna/webservices11/wscoor/client/WSCOORClient.java

```diff
@@ -41,12 +42,27 @@
     private static synchronized ActivationService getActivationService()
     {
         if (activationService.get() == null) {
-            // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
-            activationService.set(new ActivationService());
+            setActivationService();
         }
         return activationService.get();
     }
 
+    private static synchronized void setActivationService()
+    {
+        // activationService.set(new ActivationService(null, new QName("http://docs.oasis-open.org/ws-tx/wscoor/2006/06", "ActivationService")));
+        if (System.getSecurityManager() == null) {
```

Comment:
In this particular case, I don't think we can extract this privileged action, because it needs to access activationService.
However, I found a couple of places where I can extract privileged actions to their own classes and use singleton instances.